### DISCUSSION TITLE
Call to "ConfigureResolverForAsset" to generate resolver context

### DISCUSSION THREAD

**Discussion by smouajjah-sp** [Autodesk/maya-usd#1236] [Q&A]
Hello,

We have issues using resolver contexts in maya with the ADSK plugin. We sent a message to the maya beta support but didn't recieve any answer, so we try to repost the message here.

We are currently testing maya_usd 0.6.0-1 with maya 2021 beta. In our studio, we have implemented our own custom asset resolver which works well with the AL_USDMaya plugin. We tried to test it with the autodesk plugin (ADSK) and we faced some issues when importing USD files.

Indeed, our resolver uses a custom resolver context which provides some data for the asset resolution. We fill our context with data using the `ConfigureResolverForAsset` method from the resolver to generate the context. This method seems to be called by the AL_USDMaya plugin while creating a proxy shape (using the `AL_usdmaya_ProxyShapeImport` command) but never with the ADSK plugin (using the `mayaUsd_createStageFromFilePath` command).

Do you provide a way to configure the resolver properly using the new ADSK plugin so that we can use custom resolver contexts?

Thank you in advance for your help.

**Comment by kxl-adsk** [comment]
Apologies for Maya's beta post, I'm not sure how we missed it.

We have been waiting for AR 2.0 before doing any refactoring to the base proxy shape and how it handles asset resolvers. This of course doesn't address your issue at hand. I believe what's missing is a call to ConfigureResolverForAsset prior to opening the stage as it is done in here https://github.com/Autodesk/maya-usd/blob/aed181f7cb41c8ada8f9a12bce851a98532ad087/plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShape.cpp#L1325.

Is anyone else having this issue and went with a similar solution?

Cheers,
KxL

**Reply by dolivares-spin** [reply]
Yes, this is an issue for us as well. We've had similar issues in Katana and Houdini.  Usdview seems to do this automatically, and may be a good reference along with the AL_ProxyShape.  Calling ConfigureResolverForAsset before creating a mayaUsdProxyShape seems to do the trick.

**Reply by kxl-adsk** [reply]
Thank you @dolivares-spinvfx. I opened https://github.com/Autodesk/maya-usd/pull/1342 to address this.

**Reply by dolivares-spin** [reply]
Thanks for the speedy update!
